### DISCUSSION TITLE
variables defined in _variable.yml do not get passed to pandoc

### DISCUSSION THREAD

**Discussion by kompre** [quarto-dev/quarto-cli#6718] [Q&A]
### Description

So, I'm trying do define my custom titlepage for pdf output which will contains some custom variables. In order to do so I want to provide my own partial `title.tex` where I redefine the `\titlepage` command. 

During my test, I've experienced that yaml values that were defined either in the document meta block or in the `_quarto.yml` are passed to pandoc for substitution, but values defined in `_variables.yml` are not.

In the below example I defined three values, one each into `meta`, `_quarto.yml`, and `_variables.yml` respectively, then I modfied the `title.tex` partial so that I've hijacked the _title_, _subtitle_, and _author_ will display also the value defined in the custom yaml field:

- `title` -> `_variable.yml`
- `subtitile` -> `_quarto.yml`
- `author` -> `meta`

Lastly, as a sanity check, I've printed all the value above using the appropriate shortcode, to verify values are rendered by quarto.

````qmd
---
title: "banane_variables:"
subtitle: "banane_quarto"
author: autore
format:
    pdf:
        template-partials: 
            - title.tex
keep-tex: true
banane_meta: "in pigiama_meta"
---

## use pandoc with value from _variables.yml

This file test whereas the data in `_variables.yml` get picked up by pandoc when exporting to pdf.

- value in `_variables.yml`: **{{< var banane_variables >}}**
- value in `_quarto.yml` **{{< meta banane_quarto >}}**
- value in `meta` **{{< meta banane_meta >}}**

````

the `_variables.yml` looks like:

```yml
banane_variables: in pigiama_variable.yml
```

the `_quarto.yml` looks like:

```yml
project:
  title: "variables"

banane_quarto: in pigiama_quarto.yml
```

the `title-tex` partial look like:

```tex
$if(title)$
\title{title: $title$ $banana_var$}
$endif$
$if(subtitle)$
$if(beamer)$
$else$
\usepackage{etoolbox}
\makeatletter
\providecommand{\subtitle}[1]{% add subtitle to \maketitle
  \apptocmd{\@title}{\par {\large #1 \par}}{}{}
}
\makeatother
$endif$
\subtitle{subtitle: $subtitle$: $banane_quarto$}
$endif$
$if(author)$
\author{author: $banane_meta$}
$endif$

```

the pdf output looks like this:
![image](https://github.com/quarto-dev/quarto-cli/assets/11796345/f3687afd-badf-441e-88f1-5e96abb0b19e)

Only the value in `_variables.yml` is not passed to pandoc. **Is this intended behaviour?**

**Comment by dragonstyle** [comment]
I would say that this is intended behavior - the `_variables.yml` file is intended for use with the `var` shortcode rather than to define variables for Pandoc templates. Those values are not merged into the document metadata visible to Pandoc when processing the document. For that you can use `_quarto.yml` (project wide), `_metadata.yml` (folder scoped), or document front matter (document scoped).
